psr/with-let-letting.tex

```diff
@@ -0,0 +1,233 @@
+
+This PSR adds defintions for \gn{let\_clause} and \gn{letting\_clause} to figure 3 of the PartiQL specification 
+and include them it as optional clauses for \gl{SELECT} queries.
+
+\newcounter{rownum_ctr}
+\newcommand\row{\stepcounter{rownum_ctr}\arabic{rownum_ctr}}
+
+\begin{figure}[ht!]
+\centering
+\scriptsize
+\begin{tabular}{|@{~}rc@{~}l@{~}|}
+\hline
+        \row & \multicolumn{2}{@{}l@{~}|}{\gn{query}} \\
+        \row & \gp & \gn{sfw\_query} \\
+        \row & \gd & \gn{expr\_query} \\ 
+
+        \hline
+
+        \row & \multicolumn{2}{@{}l@{~}|}{\gn{sfw\_query}} \\
+        \row & \gp & (\gt{WITH} \gn{query} \gt{AS} \gn{variable})? \\
+        \row &     & \gn{select\_clause} \\ %FIXME (see Figure~\ref{figure:select:bnf}) \\
+        \row &     & \gn{from\_clause} \\   %FIXME (see Figure~\ref{figure:from:bnf}) \\ 
+        \row &     & \textbf{\gn{let\_clause}? $\leftarrow $ NEW}   \\
```

Comment:
Should surround in parens like the other optional clauses?

`(let_clause)?`

Reply:
yup.